app/src/main/java/org/literacyapp/startguide/content/swipe/SwipeUpDownActivity.java

```diff
@@ -5,31 +5,134 @@
         SwipeUpDownAdapter adapter = new SwipeUpDownAdapter(imagesArray);
 
         RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recyclerView);
         recyclerView.setLayoutManager(new LinearLayoutManager(this));
         recyclerView.setAdapter(adapter);
 
-        // TODO: 04/02/2017 play "Swipe to scroll up"
-        //MediaPlayerHelper.play(this, R.raw.);
+        //Hand view
+        mHandView = findViewById(R.id.hand);
+
+        //Add gesture detector
+        mGestureDetector = new GestureDetector(this, this);
+        recyclerView.setOnTouchListener(new View.OnTouchListener() {
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
+                return mGestureDetector.onTouchEvent(event);
+            }
+        });
+
+        showScrollUp();
+    }
+
+    /**
+     * Scroll up explanation with audio and hand animation
+     */
+    private void showScrollUp() {
+        //TODO: 04/02/2017 update audio file (en, sw) "Swipe to scroll up"
+        MediaPlayerHelper.play(this, R.raw.scroll_up);
+        AnimationHelper.animateView(this, mHandView, R.anim.slide_up);
+    }
+
+    /**
+     * Scroll down explanation with audio and hand animation
+     */
+    private void showScrollDown() {
+        //TODO: 05/02/2017 update audio file (en, sw) "Swipe to scroll down"
+        MediaPlayerHelper.play(this, R.raw.scroll_down);
+        AnimationHelper.animateView(this, mHandView, R.anim.slide_down);
+    }
+
+    private boolean isDetectScrollUpActive() {
+        return detectUp;
+    }
+
+    private boolean isDetectScrollDownActive() {
+        return detectDown;
+    }
+
+    private void setDetectUpActive(boolean activeUpDetect) {
+        detectUp = activeUpDetect;
+    }
+
+    private void setDetectDownActive(boolean activeDownDetect) {
+        detectDown = activeDownDetect;
+    }
+
+    //region GestureDetector.OnGestureListener
+    @Override
+    public boolean onDown(MotionEvent e) {
+        return false;
+    }
+
+    @Override
+    public void onShowPress(MotionEvent e) {
+    }
+
+    @Override
+    public boolean onSingleTapUp(MotionEvent e) {
+        return false;
+    }
+
+    @Override
+    public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float distanceY) {
+        return false;
+    }
+
+    @Override
+    public void onLongPress(MotionEvent e) {
+    }
+
+    @Override
+    public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX, float velocityY) {
+
+        float yMov = Math.abs(e1.getY() - e2.getY());
+        float xMov = Math.abs(e1.getX() - e2.getX());
+
+        //Detect vertical scroll
+        if (yMov > xMov) {
+
+            if (isDetectScrollUpActive() && (velocityY < 0)) {
+                //Detected scroll up
+                showScrollDown();
+
+                setDetectUpActive(false);
+                setDetectDownActive(true);
+
+            } else if (isDetectScrollDownActive() && (velocityY > 0)) {

```

Comment:
Will the navigation considered mastered after just scrolling a little bit down, or scrolling all the way to the bottom of the list?